quelpa.el

```diff
@@ -1633,13 +1632,24 @@ Return t in each case."
 
 (defun quelpa-update-cache (cache-item)
   ;; try removing existing recipes by name
-  (setq quelpa-cache (cl-remove (car cache-item)
-                                quelpa-cache :key #'car))
+  (setq quelpa-cache (--remove (eq (car cache-item) (car it))
+                               quelpa-cache))
   (push cache-item quelpa-cache)
   (setq quelpa-cache
         (cl-sort quelpa-cache #'string<
                  :key (lambda (item) (symbol-name (car item))))))
 
+(defun quelpa-remove-cache (cache-item)
```

Comment:
The name of this function suggests that it removes the cache itself rather than removing an item from the cache.